st2common/st2common/services/rules.py

```diff
@@ -0,0 +1,59 @@
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import six
+
+from st2common import log as logging
+from st2common.persistence.rule import Rule
+
+
+LOG = logging.getLogger(__name__)
+
+__all__ = [
+    'get_rules_given_trigger',
+    'get_rules_with_trigger_ref'
+]
+
+
+def get_rules_given_trigger(trigger):
+
+    if isinstance(trigger, six.string_types):
+        return get_rules_with_trigger_ref(trigger_ref=trigger)
+
+    if isinstance(trigger, dict):
+        trigger_ref = trigger.get('ref', None)
+        if trigger_ref:
+            return get_rules_with_trigger_ref(trigger_ref=trigger_ref)
+        else:
+            raise ValueError('Trigger dict %s is missing ``ref``.' % trigger)
```

Comment:
Parenthesis around `% (trigger)` would make it a bit less error prone in the future.